src/app/features/charts/components/median-barchart/median-barchart.component.spec.ts

```diff
@@ -0,0 +1,222 @@
+// -------------------------------------------------------------------------- //
+// External
+// -------------------------------------------------------------------------- //
+import { TestBed, ComponentFixture, waitForAsync } from '@angular/core/testing';
+import { RouterTestingModule } from '@angular/router/testing';
+
+// -------------------------------------------------------------------------- //
+// Internal
+// -------------------------------------------------------------------------- //
+import { MedianBarChartComponent } from './';
+import { HelperService } from '../../../../core/services';
+import { geneMock1 } from '../../../../testing';
+import { MedianExpression } from '../../../../models/genes';
+
+// -------------------------------------------------------------------------- //
+// Tests
+// -------------------------------------------------------------------------- //
+const XAXIS_LABEL = 'BRAIN REGION';
+const YAXIS_LABEL = 'LOG2CPM';
+const FULL_MOCK_DATA = geneMock1.medianexpression;
+const TISSUES = ['TCX', 'PHG', 'STG'];
+const SMALL_MOCK_DATA = [
+  {
+    ensembl_gene_id: 'MOCK_GENE',
+    minimumlogcpm: -2.54173091337051,
+    quartile1logcpm: -1.03358030635935,
+    medianlogcpm: 0.483801733963266,
+    meanlogcpm: -0.517398199667356,
+    quartile3logcpm: -0.0800759845652829,
+    maximumlogcpm: 2.32290808871289,
+    tissue: TISSUES[0],
+  },
+  {
+    ensembl_gene_id: 'MOCK_GENE',
+    minimumlogcpm: -2.44077907413711,
+    quartile1logcpm: -0.592671867557559,
+    medianlogcpm: 0.013739530502129,
+    meanlogcpm: -0.0324143336865,
+    quartile3logcpm: 0.49577213202412,
+    maximumlogcpm: 2.23019575245731,
+    tissue: TISSUES[1],
+  },
+  {
+    ensembl_gene_id: 'MOCK_GENE',
+    minimumlogcpm: -5.03189866356294,
+    quartile1logcpm: -1.02644563959975,
+    medianlogcpm: 0.176348063122062,
+    meanlogcpm: -0.323038107200895,
+    quartile3logcpm: 0.391874711168331,
+    maximumlogcpm: 1.9113258251877,
+    tissue: TISSUES[2],
+  },
+];
+
+describe('Component: BarChart - Median', () => {
+  let fixture: ComponentFixture<MedianBarChartComponent>;
+  let component: MedianBarChartComponent;
+  let element: HTMLElement;
+
+  beforeEach(waitForAsync(() => {
+    TestBed.configureTestingModule({
+      declarations: [MedianBarChartComponent],
+      imports: [RouterTestingModule],
+      providers: [HelperService],
+    }).compileComponents();
+  }));
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(MedianBarChartComponent);
+    component = fixture.componentInstance;
+  });
+
+  const setUp = (
+    data: MedianExpression[] = FULL_MOCK_DATA,
+    xAxisLabel: string = XAXIS_LABEL,
+    yAxisLabel: string = YAXIS_LABEL
+  ) => {
+    component.data = data;
+    component.xAxisLabel = xAxisLabel;
+    component.yAxisLabel = yAxisLabel;
+
+    fixture.detectChanges();
+    element = fixture.nativeElement;
+    const chart = element.querySelector('svg g');
```

Comment:
Since the svg exists in the template and so may be present when the chart hasn't been created, use the first child added in `createChart` to check whether the chart exists